### PR TITLE
Kconfig cleanup - logging

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

**Review by nashif — APPROVED** [review]

File: drivers/modem/modem_receiver.c [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 #include <logging/log.h>
 #define LOG_DOMAIN mdm_receiver
-#define LOG_LEVEL CONFIG_LOG_MODEM_LEVEL
+#define LOG_LEVEL CONFIG_MODEM_LOG_LEVEL
 LOG_MODULE_REGISTER(LOG_DOMAIN);
```
Comment (nordic-krch) [inline drivers/modem/modem_receiver.c]: those 3 lines may be converted to:
`LOG_MODULE_REGISTER(mdm_receiver, CONFIG_MODEM_LOG_LEVEL);`
this has the advantage that not existing defines would be reported.

File: drivers/pwm/pwm_sifive.c [hunk 2]
```diff
@@ -4,7 +4,7 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#define LOG_LEVEL CONFIG_LOG_PWM_LEVEL
+#define LOG_LEVEL CONFIG_PWM_LOG_LEVEL
```
Comment (nordic-krch) [inline drivers/pwm/pwm_sifive.c]: ditto

File: subsys/disk/disk_access_sdhc.c [hunk 3]
```diff
@@ -4,7 +4,7 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#define LOG_LEVEL CONFIG_DISC_LOG_LEVEL
+#define LOG_LEVEL CONFIG_DISK_LOG_LEVEL
 #include <logging/log.h>
 LOG_MODULE_REGISTER(sdhc);
```
Comment (nordic-krch) [inline subsys/disk/disk_access_sdhc.c]: `LOG_MODULE_REGISTER(sdhc, CONFIG_DISK_LOG_LEVEL);`

**Review by nordic-krch — CHANGES_REQUESTED** [review]

File: drivers/modem/modem_receiver.c (hunk 1 above)
Comment (galak) [reply drivers/modem/modem_receiver.c]: Fixed

File: subsys/disk/disk_access_sdhc.c (hunk 3 above)
Comment (galak) [reply subsys/disk/disk_access_sdhc.c]: Fixed.

File: drivers/pwm/pwm_sifive.c (hunk 2 above)
Comment (galak) [reply drivers/pwm/pwm_sifive.c]: Fixed.

**Comment by galak** [comment]
@nordic-krch made the changes, take another look.

**Review by nordic-krch — APPROVED** [review]

**Review by pizi-nordic — APPROVED** [review]

**Pull request merged**